### PR TITLE
Updating release.md to current workflow

### PR DESCRIPTION
**Pull request by patflynn**

### CODE REVIEW

**Comment by loosebazooka** [comment]
Can you pull this down to your computer and recommit so it uses the correct email address?

**Comment by patflynn** [comment]
@loosebazooka done.

File: RELEASING.md [hunk 1]
```diff
@@ -2,45 +2,18 @@
 
 This page describes how to cut releases of our plugin. It's strictly for the owners of the project.
 
-1. Create a new branch as if you were starting a normal bugfix PR
+1. Create a new branch named release_v{$version}.  It's critical that you create the branch according to this schema or the release will fail.
```
Comment (loosebazooka) [inline RELEASING.md]: Is there concern local branches wont be up to date?

how about something like?

```
git fetch
git checkout -b release_v1.0 origin/master
```

Comment (patflynn) [inline RELEASING.md]: The release will fail if the local branch isn't up to date, but fair point
that the instructions should help the user make sure they are up to date
before they start.

Comment (patflynn) [inline RELEASING.md]: done.

**Comment by patflynn** [comment]
Any more questions about this change?

**Comment by loosebazooka** [comment]
lgtm

**Pull request merged**